### DISCUSSION TITLE
Functions?

### DISCUSSION THREAD

**Discussion by ddebowczyk** [spatie/php-structure-discoverer#21] [Q&A]
Is there a way to discover functions as an additional DiscoveredStructureType?

**Comment by rubenvanassche** [comment]
Not at the moment, what would be your usecase?

**Reply by ddebowczyk** [reply]
I'm investigating using Structure Discoverer for auto-documenting the codebase.

**Reply by rubenvanassche** [reply]
Why not discover classes and then use reflection to get the methods?

**Reply by ddebowczyk** [reply]
I'm asking about functions, not methods.

**Reply by rubenvanassche** [reply]
Nowp, no support

**Reply by ddebowczyk** [reply]
Thanks for the answer. Does "no support" mean "no - currently" or "no and never"?

**Reply by rubenvanassche** [reply]
Not currently, also no plans to add this feature but you're always welcome to send in a PR

**Reply by ddebowczyk** [reply]
OK, thanks!
